cmd/podman/common/create.go

```diff
@@ -1053,3 +1057,46 @@ func DefineCreateFlags(cmd *cobra.Command, cf *entities.ContainerCreateOptions,
 	)
 	_ = cmd.RegisterFlagCompletionFunc(blkioWeightDeviceFlagName, completion.AutocompleteDefault)
 }
+
+func GetHealthCheckOverrideConfig(cmd *cobra.Command, vals *entities.ContainerCreateOptions) (*manifest.Schema2HealthConfig, error) {
```

Comment:
I don't understand the need for a new function. IMO it should also just be in MakeHealthCheckFromCli() and it needs to be fixed there to always work when only one option is set.

Reply:
I think this is problematic because the `FillOutSpecGen()` function that calls the `MakeHealthCheckFromCli()` function has no idea which flags have been changed, so it uses the default or command line value. Here, we have no option to determine whether the user has set a default value or whether it is the default value with no change.  And the values specified by the flags take precedence, so they override the configuration from the image.

Reply:
Then the command line should not pass in any default and just empty strings I would say.

This whole thing of the command line sets the default must stop long term anyway IMO. It causes nothing but issues like this.

Reply:
So can I change the default flag settings? Is this an allowed API change? Or reset them to empty values when they are not used in the command line?

Reply:
Yeah I would leave the cli default value empty and then set the default at a later stage, i.e. on the server.
The only thing about the setting is on the client is that podman run --help shows the default value. IMO that is not worth the troubles it has caused us.